### PR TITLE
fix(identity): parse managed identities

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- add managed identity name parsing for agent identities
- fall back to managed identity parsing when ziti management returns NotFound
- cover managed identity parsing with unit tests

## Testing
- buf generate buf.build/agynio/api --path agynio/api/llm/v1 --path agynio/api/users/v1 --path agynio/api/authorization/v1 --path agynio/api/ziti_management/v1 --path agynio/api/identity/v1
- go vet ./...
- go build ./...
- go test ./...

#43

### CODE REVIEW

File: internal/identity/managed_identity_test.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package identity
+
+import "testing"
+
+func TestParseManagedIdentityName(t *testing.T) {
```
Comment (noa-lucent) [inline internal/identity/managed_identity_test.go]: [major] The test suite is missing critical edge cases for the `LastIndex`-based split logic. Please add cases for:

- `"agent-"` — prefix only, no UUID, no suffix.
- `"agent-9e2b5e44-965a-4a6d-a1ea-2cf9b8b49ab2"` — valid UUID but **no suffix**. `LastIndex` would split *inside* the UUID (at the last `-` before `2cf9b8b49ab2`), producing an invalid UUID fragment. This is the subtlest behavior of the parser and must be covered.
- `"  agent-9e2b5e44-965a-4a6d-a1ea-2cf9b8b49ab2-abc  "` — leading/trailing whitespace (verifying `TrimSpace` works end-to-end).

Also, consider converting to a table-driven test (`[]struct{ name, input string; wantOK bool; wantID string }`) — it's more idiomatic Go and makes it trivial to add more cases.
